cub/cub/util_namespace.cuh

```diff
@@ -49,6 +49,8 @@
 
 #include <cub/detail/detect_cuda_runtime.cuh>
 
+#include <thrust/detail/config/namespace.h>
+
```

Comment:
Does cub depend on thrust or the other way around?

Reply:
They depend on each other in both directions. Thrust only uses CUB when the CUDA backend is on. CUB uses Thrust in all cases for kernel launches and unwrapping of iterators/device_references.

I wasn't happy with cross including here as well. The alternative is to move the common splicing macro somewhere into libcu++ I guess. Any suggestions?

Reply:
Given that this macro is only used in CUDA backend of Thrust, I'd rather make Thrust CUDA backend use CUB macro than other way around. Ideally, this macro might be needed on the cudax end, so I'd consider hosting it to `_CCCL_` level.

Reply:
I asked @miscco for a good place inside libcu++ and we couldn't find a quick answer. Let's try again.

Reply:
We will put it into `__cccl/preprocessor.h`